zanata-war/src/main/java/org/zanata/dao/ApplicationConfigurationDAO.java

```diff
@@ -1,12 +1,14 @@
 package org.zanata.dao;
 
 import org.hibernate.Session;
+
+import javax.enterprise.context.RequestScoped;
 import javax.inject.Named;
 import org.zanata.model.HApplicationConfiguration;
 
 @Named("applicationConfigurationDAO")
 
-@javax.enterprise.context.Dependent
+@RequestScoped
```

Comment:
Because EntityManager and Session are Request scope. DAO makes sense to
stay in the same scope.

On Fri, Jan 22, 2016 at 10:17 AM, Carlos Munoz <notifications@github.com>
wrote:

> In
> zanata-war/src/main/java/org/zanata/dao/ApplicationConfigurationDAO.java
> <https://github.com/zanata/zanata-server/pull/1084#discussion_r50486499>:
>
> >  import javax.inject.Named;
> >  import org.zanata.model.HApplicationConfiguration;
> >
> >  @Named("applicationConfigurationDAO")
> >
> > -@javax.enterprise.context.Dependent
> > +@RequestScoped
>
> I think you're right... all other DAOs are request scoped. In light of
> this I remove my objection [image: :wink:]
>
> @seanf <https://github.com/seanf> @huangp <https://github.com/huangp> Do
> you remember why we had to make them Request scoped instead of dependent?
> Proxies perhaps?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1084/files#r50486499>.
>



-- 
Patrick Huang
Senior Software Engineer
Engineering - Internationalisation
Red Hat, Asia-Pacific Pty Ltd
Level 1, 193 North Quay
Brisbane 4000
Office: +61 7 3514 8278
Fax: +61 7 3514 8199
IRC: pahuang
github: github.com/huangp
Website: www.redhat.com